### PR TITLE
fix(auth0): add types for child classes

### PR DESCRIPTION
**Pull request by wbhob**
Please fill in this template.

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `tsc` without errors.
- [X] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/auth0/node-auth0
- [X] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

**Comment by westy92** [comment]
Please add some tests as well.

File: auth0/index.d.ts [hunk 1]
```diff
@@ -217,6 +217,58 @@ export class AuthenticationClient {
...
+    sendEmail(userData: { email: string; send: 'link' | 'code' }, cb: (err: Error, message: string) => void): void;
+
+    sendSMS(userData: { phone_number: string; client_id?: string }): Promise<any>;
+    sendSMS(userData: { phone_number: string; client_id?: string }, cb: (err: Error, message: string) => void): void;
+
+
+  };
+  users: {
+
+    getInfo(userData: { accessToken: string }): Promise<any>;
```
Comment (westy92) [inline auth0/index.d.ts]: I know this wasn't done before, but can we add stricter return types than `Promise<any>`?

**Comment by westy92** [comment]
While you're making changes, could you quickly fix https://github.com/auth0/node-auth0/issues/169?

**Comment by wbhob** [comment]
I frankly don't know how to add tests, and I have no idea what the methods are returning.

**Comment by westy92** [comment]
The "tests" are [here](https://github.com/wbhob/DefinitelyTyped/blob/master/auth0/auth0-tests.ts).  Add some usage of these new typings.

As far as the return values, I guess you can skip them for now.

Thank you for the contribution!

**Comment by paulvanbrenk** [comment]
Thanks for the contribution, I agree with the request for the tests (they only need to be a couple of lines).

**Comment by wbhob** [comment]
Guh that's going to take an eternity. I'll get on it when I get the opportunity

**Comment by andy-ms** [comment]
We've recently updated the repository structure.
All packages are now in a `types` subdirectory.
Please rebase this pull request onto the latest `master` branch.

**Comment by RyanCavanaugh** [comment]
Please fix merge conflicts. Thanks!

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has failing CI tests and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**